frontend-ssr/redux/exercise-submissions/exercise-submissions.actions.ts

```diff
@@ -0,0 +1,30 @@
+import { Submission } from './types';
+
+const ADD_SUBMISSION = 'exercise-submissions/ADD';
+const REMOVE_SUBMISSION = 'exercise-submissions/REMOVE';
+const UPDATE_SUBMISSION = 'exercise-submissions/UPDATE';
+
+export const EXERCISE_SUBMISSIONS = {
+  ADD_SUBMISSION,
+  REMOVE_SUBMISSION,
+  UPDATE_SUBMISSION,
+};
+
+export const addSubmission = (submission: Submission, index: number) => ({
+  type: ADD_SUBMISSION,
+  payload: {
+    submission,
+    index,
+  },
+});
+export const removeSubmission = (id: string) => ({
```

Comment:
Same here. If we support single or multiple, types should be `string | string[]`.